### PR TITLE
Workaround for Windows DPI issues

### PR DESCRIPTION
**Pull request by felixrieseberg**
This isn't really a solution, this is more a workaround in case we can't find any real solution. 

On Windows, multiple screens with different scaling factors result in the _first_ call to `setBounds` being incorrect. Subsequent calls will place the window correctly. This dirty fix simply does that.

Ref https://github.com/electron/electron/issues/10862

### CODE REVIEW

**Comment by alespergl** [comment]
This definitely gets points for its embedded quality of invoking the spirits. :)
Yeah, why not. If it _really_ helps... Doesn't seem like it could hurt anyway.

BTW I noticed today that some unit tests fail when executed in a high-DPI context. Fixing those tests might actually help us find the right solutions to these types of problems. I thought I would have a look at that at some point in near future after we're done with the current Chromium upgrade.

File: atom/browser/native_window.cc [hunk 1]
```diff
@@ -117,6 +117,14 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
   bool center;
   if (options.Get(options::kX, &x) && options.Get(options::kY, &y)) {
     SetPosition(gfx::Point(x, y));
+
+#if DEFINEED(OS_WIN)
```
Comment (mgc) [inline atom/browser/native_window.cc]: defineEd?

File: atom/browser/native_window.cc [hunk 2]
```diff
@@ -117,6 +117,14 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
   bool center;
   if (options.Get(options::kX, &x) && options.Get(options::kY, &y)) {
     SetPosition(gfx::Point(x, y));
+
+#if DEFINED(OS_WIN)
```
Comment (poiru) [inline atom/browser/native_window.cc]: I think this needs to be lowercase `defined`.

File: atom/browser/native_window.cc [hunk 3]
```diff
@@ -117,6 +117,14 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
   bool center;
   if (options.Get(options::kX, &x) && options.Get(options::kY, &y)) {
     SetPosition(gfx::Point(x, y));
+
+#if defined(OS_WIN)
+    // Dirty, dirty workaround for
```
Comment (zcbenz) [inline atom/browser/native_window.cc]: Can you add `FIXME(felixrieseberg): ` in the comment?

**Comment by bpasero** [comment]
To avoid calling `SetPosition` twice in cases where not needed, wouldn't it be better to immediately after `SetPosition` ask the window for its position and only call `SetPosition` again if the position is different? Also it seems that the code should not run when just one monitor is attached.

**Comment by zcbenz** [comment]
> To avoid calling SetPosition twice in cases where not needed, wouldn't it be better to immediately after SetPosition ask the window for its position and only call SetPosition again if the position is different? Also it seems that the code should not run when just one monitor is attached.

Setting window position is a rather cheap operation before window is shown, and it is only called twice when window is being created, so I think it is fine do it blindly here.

**Comment by bpasero** [comment]
@zcbenz ok fine 👍

**Pull request merged**